onnxruntime/python/tools/transformers/fusion_reshape.py

```diff
@@ -31,16 +48,43 @@ def fuse(self, reshape_node, input_name_to_nodes, output_name_to_node):
 
         (unsqueeze_0, gather_0, shape_0) = path0
 
+        shape = []
+        gather_value = self.model.get_constant_value(gather_0.input[1])
+        if gather_value == 0:
+            shape.append(0)
+
         path1 = self.model.match_parent_path(concat_node, ['Unsqueeze', 'Gather', 'Shape'], [1, 0, 0],
                                              output_name_to_node)
         if path1 is None:
+            # Adjust for Bart
```

Comment:
These reshape fusion is not generic. I think it is better to put to preprocess of onnx_model_bart.py.

Reply:
This reshape fusion is based on the assumption that hidden_state is >2d and gemm won't change the first and second dimension. So I think it's generic to transformer model?